src/plugins/ui_feature/renderer.ts

```diff
@@ -204,32 +261,113 @@ export class RendererPlugin extends UIPlugin {
 
   private drawBorders(renderingContext: GridRenderingContext) {
     const { ctx, thinLineWidth } = renderingContext;
+    const drawBorder = ([style, color], x1, y1, x2, y2) => {
+      const lineWidth = (style === "thin" ? 1 : 2) * thinLineWidth;
+      this.drawSingleGridLine(ctx, color, lineWidth, { x1, y1, x2, y2 });
+    };
     for (let box of this.boxes) {
       const border = box.border;
       if (border) {
         const { x, y, width, height } = box;
         if (border.left) {
-          drawBorder(border.left, x, y, x, y + height);
+          drawBorder(border.left, x, y - thinLineWidth / 2, x, y + height + thinLineWidth / 2);
         }
         if (border.top) {
-          drawBorder(border.top, x, y, x + width, y);
+          drawBorder(border.top, x - thinLineWidth / 2, y, x + width + thinLineWidth / 2, y);
         }
         if (border.right) {
           drawBorder(border.right, x + width, y, x + width, y + height);
         }
         if (border.bottom) {
-          drawBorder(border.bottom, x, y + height, x + width, y + height);
+          drawBorder(
+            border.bottom,
+            x - thinLineWidth / 2,
+            y + height,
+            x - thinLineWidth / 2 + width,
+            y + height
+          );
         }
       }
     }
+  }
 
-    function drawBorder([style, color], x1, y1, x2, y2) {
+  private drawSingleGridLine(
+    ctx: CanvasRenderingContext2D,
+    color: string,
+    lineWidth: number,
+    {
+      x1,
+      y1,
+      x2,
+      y2,
+    }: {
+      x1: number;
+      y1: number;
+      x2: number;
+      y2: number;
+    }
```

Comment:
kind of a detail, but I'd be probably better if you defined the interface ``{x1 : number, y1 : ....}` somewhere else, having the arguments of the function spread on 8 lines is IMHO ulgy and unreadable.

IMO if unpacking an argument takes more than one line because of prettier, we should just unpack in the first line of the funciton instead, I think its more readable.